src/scene/batching.js

```diff
@@ -0,0 +1,719 @@
+pc.extend(pc, function () {
+
+    var Batch = function (meshInstances, isDynamic) {
+        this.origMeshInstances = meshInstances;
+        this._aabb = new pc.BoundingBox();
+        this.meshInstance = null;
+        this.model = null;
+        this.isDynamic = isDynamic;
+    };
+
+    var BatchGroup = function (id, name, isDynamic, maxAabbSize) {
+        this.isDynamic = isDynamic;
+        this.maxAabbSize = maxAabbSize;
+        this.id = id;
+        this.name = name;
+    };
+
+    // Modified SkinInstance for batching
+    // Doesn't contain bind matrices, simplier
+    var SkinBatchInstance = function (device, nodes, rootNode) {
```

Comment:
Is this actually used to do Skinning or is it just that the technique is similar? If it is just a similar idea, then I would remove the term Skin from it all because it is confusing.

Reply:
It's a simplified copy of SkinInstance. Engine treats it like a normal skin instance, but it doesn't do matrix multiplies.